include IVIM code from Dr. Amita Shukla-Dave Lab at MSKCC

locastre opened the pull request:
We have contributed our codebase for intravoxel incoherent motion (IVIM) code from the MRI-QAMPER MATLAB package, developed by Dr. Amita Shukla-Dave's lab at Memorial Sloan Kettering Cancer Center.

__Authors:__ Eve LoCastro (locastre@mskcc.org), Dr. Ramesh Paudyal (paudyalr@mskcc.org), Dr. Amita Shukla-Dave (davea@mskcc.org) 
__Institution:__ Memorial Sloan Kettering Cancer Center
__Department:__ Medical Physics
__Address:__ 321 E 61st St, New York, NY 10022

etpeterson commented on src/original/ASD_MemorialSloanKettering/MRI-QAMPER_IVIM/utils/nifti_toolbox/license.txt:
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2014, Jimmy Shen
```
Should this be included? It seems to be a separate toolbox. https://www.mathworks.com/matlabcentral/fileexchange/8797-tools-for-nifti-and-analyze-image

oliverchampion replied:
Hey! Thanks for this great submit! 
Some questions:
It seems like the toolbox includes some generic toolboxes like the NIFTI toolbox. Do we need that to be part of the package, or could we take it as a requirement?
Also, what does the code do? Is it just fitting, or does it do more processing?

Thanks!

Oliver and Eric

locastre replied:
@oliverchampion @etpeterson 

The separate nifti_toolbox is relied on extensively to handle nifti input and output in our MATLAB code. This toolbox is freely available on MATLAB FileExchange and as such is subject to the BSD License with the following stipulations:

```Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
```

Cited as: Jimmy Shen (2025). Tools for NIfTI and ANALYZE image (https://www.mathworks.com/matlabcentral/fileexchange/8797-tools-for-nifti-and-analyze-image), MATLAB Central File Exchange. Retrieved April 24, 2025.

locastre commented:
Hi Oliver and Eric,

Sure, it is fine to list nifti_toolbox as a requirement. The code does some optional image smoothing for preprocessing and accounting for SNR. It also checks that the bvalue images are in the correct (increasing b-value) order. It outputs the ROI parameter maps for D, D*, and f, and also outputs ROI-average values into a CSV file. I can add more of this information into the script comments as documentation.


Cheers,
Eve

Eve LoCastro, M.S.
Advanced Software Developer
Medical Physics, Service for Predictive Informatics


Memorial Sloan Kettering Cancer Center<https://mskcc.org/>
321 E 61st St, Rm. 207C
New York, NY 10022

Mobile: +1 (718) 406-4248
***@***.******@***.***>

From: Oliver Gurney-Champion ***@***.***>
Sent: Wednesday, March 6, 2024 11:23 AM
To: OSIPI/TF2.4_IVIM-MRI_CodeCollection ***@***.***>
Cc: LoCastro, Eve ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [OSIPI/TF2.4_IVIM-MRI_CodeCollection] include IVIM code from Dr. Amita Shukla-Dave Lab at MSKCC (PR #52)

@oliverchampion commented on this pull request. In src/original/ASD_MemorialSloanKettering/MRI-QAMPER_IVIM/utils/nifti_toolbox/license. txt: > @@ -0,0 +1,24 @@ +Copyright (c) 2014, Jimmy Shen Hey! Thanks for this great submit! Some questions: 


@oliverchampion commented on this pull request.

________________________________

In src/original/ASD_MemorialSloanKettering/MRI-QAMPER_IVIM/utils/nifti_toolbox/license.txt<https://urldefense.com/v3/__https:/github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52*discussion_r1514784620__;Iw!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCIGShdOg$>:

> @@ -0,0 +1,24 @@

+Copyright (c) 2014, Jimmy Shen

Hey! Thanks for this great submit!
Some questions:
It seems like the toolbox includes some generic toolboxes like the NIFTI toolbox. Do we need that to be part of the package, or could we take it as a requirement?
Also, what does the code do? Is it just fitting, or does it do more processing?

Thanks!

Oliver and Eric

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52*discussion_r1514784620__;Iw!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCIGShdOg$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AID5QMAL2XXJ57CQFLJFKBTYW47GNAVCNFSM6AAAAABEF7KHU2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRQGIZTANZZGE__;!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCMN5yy_8$>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

=====================================================================

Please note that this e-mail and any files transmitted from
Memorial Sloan Kettering Cancer Center may be privileged, confidential,
and protected from disclosure under applicable law. If the reader of
this message is not the intended recipient, or an employee or agent
responsible for delivering this message to the intended recipient,
you are hereby notified that any reading, dissemination, distribution,
copying, or other use of this communication or any of its attachments
is strictly prohibited. If you have received this communication in
error, please notify the sender immediately by replying to this message
and deleting this message, any attachments, and all copies and backups
from your computer.

Disclaimer ID:MSKCC

locastre commented:
Thanks so much, Eve
Amita

Amita Shukla-Dave, Ph. D., FISMRM
Vice Chairman Radiology-Medical Physics Outreach
Deputy Service Chief Predictive Informatics
Director Quantitative Imaging
Attending Physicist (Professor), Member
Departments of Medical Physics and Radiology
Memorial Sloan-Kettering Cancer Center
New York, NY 10065
phone:212-639-3184
***@***.***
http://www.mskcc.org/research/lab/amita-dave



From: LoCastro, Eve ***@***.***>
Sent: Wednesday, March 06, 2024 11:28 AM
To: OSIPI/TF2.4_IVIM-MRI_CodeCollection ***@***.***>; OSIPI/TF2.4_IVIM-MRI_CodeCollection ***@***.***>
Cc: Dave, Amita ***@***.***>
Subject: RE: [EXTERNAL] Re: [OSIPI/TF2.4_IVIM-MRI_CodeCollection] include IVIM code from Dr. Amita Shukla-Dave Lab at MSKCC (PR #52)

Hi Oliver and Eric,

Sure, it is fine to list nifti_toolbox as a requirement. The code does some optional image smoothing for preprocessing and accounting for SNR. It also checks that the bvalue images are in the correct (increasing b-value) order. It outputs the ROI parameter maps for D, D*, and f, and also outputs ROI-average values into a CSV file. I can add more of this information into the script comments as documentation.


Cheers,
Eve

Eve LoCastro, M.S.
Advanced Software Developer
Medical Physics, Service for Predictive Informatics


Memorial Sloan Kettering Cancer Center<https://mskcc.org/>
321 E 61st St, Rm. 207C
New York, NY 10022

Mobile: +1 (718) 406-4248
***@***.******@***.***>

From: Oliver Gurney-Champion ***@***.******@***.***>>
Sent: Wednesday, March 6, 2024 11:23 AM
To: OSIPI/TF2.4_IVIM-MRI_CodeCollection ***@***.******@***.***>>
Cc: LoCastro, Eve ***@***.******@***.***>>; Author ***@***.******@***.***>>
Subject: [EXTERNAL] Re: [OSIPI/TF2.4_IVIM-MRI_CodeCollection] include IVIM code from Dr. Amita Shukla-Dave Lab at MSKCC (PR #52)

@oliverchampion commented on this pull request. In src/original/ASD_MemorialSloanKettering/MRI-QAMPER_IVIM/utils/nifti_toolbox/license. txt: > @@ -0,0 +1,24 @@ +Copyright (c) 2014, Jimmy Shen Hey! Thanks for this great submit! Some questions: 


@oliverchampion commented on this pull request.

________________________________

In src/original/ASD_MemorialSloanKettering/MRI-QAMPER_IVIM/utils/nifti_toolbox/license.txt<https://urldefense.com/v3/__https:/github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52*discussion_r1514784620__;Iw!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCIGShdOg$>:

> @@ -0,0 +1,24 @@

+Copyright (c) 2014, Jimmy Shen

Hey! Thanks for this great submit!
Some questions:
It seems like the toolbox includes some generic toolboxes like the NIFTI toolbox. Do we need that to be part of the package, or could we take it as a requirement?
Also, what does the code do? Is it just fitting, or does it do more processing?

Thanks!

Oliver and Eric

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52*discussion_r1514784620__;Iw!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCIGShdOg$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AID5QMAL2XXJ57CQFLJFKBTYW47GNAVCNFSM6AAAAABEF7KHU2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRQGIZTANZZGE__;!!KVWo1iE!SMb7JOGD6xXObPLSfLIkvbFVOfkf4pGmz3lKKF1U9pLihSeHCRtNUd9jtx45w9Coa1hWbs53beshAisCMN5yy_8$>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

=====================================================================

Please note that this e-mail and any files transmitted from
Memorial Sloan Kettering Cancer Center may be privileged, confidential,
and protected from disclosure under applicable law. If the reader of
this message is not the intended recipient, or an employee or agent
responsible for delivering this message to the intended recipient,
you are hereby notified that any reading, dissemination, distribution,
copying, or other use of this communication or any of its attachments
is strictly prohibited. If you have received this communication in
error, please notify the sender immediately by replying to this message
and deleting this message, any attachments, and all copies and backups
from your computer.

Disclaimer ID:MSKCC

etpeterson commented:
Just checking in on this because it's been a while. Were you planning on cleaning it or were you expecting someone to take this over?

oliverchampion commented:
Hey! Way too late, but I am reviewing the code. You code is way more extensive then all the other commits, so this may take some more effort/time. But we will get there!
For now there are some things to address.

oliverchampion commented:
You have a demo as "demo_QAMPER_IVIM.m", which is very useful. But as Git gets clogged up fast with files, we do not want nii files in Git. I see you have removed the test data (good!) but obviously the demo now fails. 

Luckily, our Git comes with some test data that is stored on Zenodo: https://zenodo.org/records/10696605. This data is automatically downloaded whenever needed, as we coded the git such that it runs utilities/data_simulation/Download_data.py and stores it in "download\Data". I am now running your code with the abdomen.nii.gz data, and this works! However, it required me to (1) download the data manually and (2) make a mask.

So my questions are:
1: can we make sure the data is downloaded automatically whenever we need it for you demo (maybe this is not possible... Then you can refer to the location in the file)
2: is it required to add a mask.nii.gz or can it run without a mask/an automated mask?

oliverchampion commented:
I think there are several files in here that are not needed, such as the .asv files. Could you remove the asv files, and add .asv to the .gitignore?

oliverchampion commented:
> Apologies for the late responses on my part, I have not been receiving the notifications for this thread. I'll work on addressing the ASV and demo scripts now, sending update when they are complete.

Any updates :)?

locastre commented:
Dear @oliverchampion and @etpeterson, thanks for your feedback and patience. I've tried to address all of your questions with several commits this past week. 

- Remove `*.asv` files and create a `.gitignore` file to address that pattern for future commits
- Remove unneeded files that refer to other processing modalities (ie, DCE, T1 FA, etc)
- Provide a script to download the OSIPI Zenodo demo data and include it in our `demo_QAMPER_IVIM.m` script
- If no ROI is provided by the user, the program will default to calculate IVIM parameters for all voxels in the volume.

Please let us know if there are any other issues we should address before merging!

etpeterson reviewed (approved):
Thanks, I think this is good to merge. I'll let Oliver do a check before merging though.

locastre commented:
Dear Oliver,

Thank you for your feedback and prompting to complete our submission to the OSIPI IVIM code repository. I think we have addressed all of the comments from GitHub. Here is the rundown of our updates (https://github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52#issuecomment-2832714733)



  *   Remove *.asv files and create a .gitignore file to address that pattern for future commits
  *   Remove unneeded files that refer to other processing modalities (ie, DCE, T1 FA, etc)
  *   Provide a script to download the OSIPI Zenodo demo data and include it in our demo_QAMPER_IVIM.m script
  *   If no ROI is provided by the user, the program will default to calculate IVIM parameters for all voxels in the volume.


Please let us know if there are any other issues we should address before merging!

When it becomes available, will you please send us the link for the manuscript?


Best regards,
Eve

Eve LoCastro, M.S.
Advanced Software Developer
Medical Physics, Service for Predictive Informatics


Memorial Sloan Kettering Cancer Center<https://mskcc.org/>
321 E 61st St, Rm. 207C
New York, NY 10022

Mobile: +1 (718) 406-4248
***@***.******@***.***>

From: Eric Peterson ***@***.***>
Sent: Sunday, April 27, 2025 12:53 AM
To: OSIPI/TF2.4_IVIM-MRI_CodeCollection ***@***.***>
Cc: LoCastro, Eve ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [OSIPI/TF2.4_IVIM-MRI_CodeCollection] include IVIM code from Dr. Amita Shukla-Dave Lab at MSKCC (PR #52)

@ etpeterson approved this pull request. Thanks, I think this is good to merge. I'll let Oliver do a check before merging though. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were


@etpeterson approved this pull request.

Thanks, I think this is good to merge. I'll let Oliver do a check before merging though.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/pull/52*pullrequestreview-2797129382__;Iw!!KVWo1iE!UVwijRZdK34t2KElUePb5jT2D-Ra2aAuJhD4raReaG98imBfSMWSBFYrEn-1sf6NzJfwbnnaT_alTgXug8hudTM$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AID5QMBHYLVQ2X2VZGBWIYD23RPCBAVCNFSM6AAAAAB3ZEUYN2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOOJXGEZDSMZYGI__;!!KVWo1iE!UVwijRZdK34t2KElUePb5jT2D-Ra2aAuJhD4raReaG98imBfSMWSBFYrEn-1sf6NzJfwbnnaT_alTgXuNKrQU_I$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

=====================================================================

Please note that this e-mail and any files transmitted from
Memorial Sloan Kettering Cancer Center may be privileged, confidential,
and protected from disclosure under applicable law. If the reader of
this message is not the intended recipient, or an employee or agent
responsible for delivering this message to the intended recipient,
you are hereby notified that any reading, dissemination, distribution,
copying, or other use of this communication or any of its attachments
is strictly prohibited. If you have received this communication in
error, please notify the sender immediately by replying to this message
and deleting this message, any attachments, and all copies and backups
from your computer.

Disclaimer ID:MSKCC

oliverchampion commented:
Hey! 
I've taken the liberty to change some aspects:
- You've removed the default git-ignore and created one yourself. This means a lot of ignored files will be merged. Fixed this
- There were some typo's in the matlab file (fulllfile with a L too much)
- By re-downloading the data to a new location, the data was downloaded twice. I've now pointed to the default data location in the Git, such that the data is shared.
